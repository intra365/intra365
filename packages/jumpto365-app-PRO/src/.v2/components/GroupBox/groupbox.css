.gb0 {
    transform-origin: 0% 0% 0%;
    transform: rotate(0deg)
}

.gb90 {
    transform-origin: left top ;

    transform: rotate(90deg)
    
}

.gb180 {
    transform-origin: 100% 100% 0%;
    transform: rotate(180deg)
}

.gb270 {
    transform-origin: left  top ;
    transform: rotate(270deg)
}

.gbText{
    flex-wrap: wrap;
    text-align: center;
    font-weight: 200;
    text-transform: uppercase;
    text-indent: 3%
}

.groupboxeditor:hover{
    background-color: #eeeeee
}