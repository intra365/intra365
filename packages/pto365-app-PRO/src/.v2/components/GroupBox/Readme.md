


## 0
```js
<GroupBox text="Fájltárolás és Együttműködés" color="green"  rotation="0" height="55px" width="300px" fontsize="12px"/>
```

## 90
```js
<GroupBox text="Fájltárolás és Együttműködés" color="red"  rotation="90"  height="55px" width="300px" fontsize="12px"/>
```

## 180
```js
<GroupBox text="Fájltárolás és Együttműködés" color="yellow"  rotation="180"  height="55px" width="300px" fontsize="12px"/>
```

## 270
```js
<GroupBox text="Fájltárolás és Együttműködés" color="orange"  rotation="270"  height="55px" width="300px" fontsize="12px"/>
```