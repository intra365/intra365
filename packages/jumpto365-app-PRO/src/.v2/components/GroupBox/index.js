import React, { Component } from "react";
import PropTypes from "prop-types";
import "./groupbox.css";
import { Provider, Subscribe } from "react-contextual";
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class GroupBox
 * @extends {Component}
 */
export default class GroupBox extends Component {
  state = {};

  static propTypes = {
    text: PropTypes.string,
    fontsize: PropTypes.string,
    height: PropTypes.string,
    width: PropTypes.string,
    color: PropTypes.string,
    textColor: PropTypes.string,
    rotation: PropTypes.string //supported - 0,90,180,270
  };

  constructor(props) {
    super(props);
  }

  /**
   * Required method return the output of the component
   *
   * @returns%
   * @memberof GroupBox
   */
  render() {
    var height = this.props.height;
    var width = this.props.width;
    var className = "gb0";
    var marginLeft = 0;
    var marginTop = 0;

    var onGroupClicked =
      this.props.editMode && this.props.editor
        ? this.props.editor.onGroupClicked
        : null;
var pencilTop = ""
    switch (this.props.rotation ? this.props.rotation.toUpperCase() : "") {
      case "90":
      pencilTop = `3px`
      height = this.props.width;
        width = this.props.height;
        className = "gb90";
        marginLeft = width;
        break;
      case "180":
        className = "gb180";

        break;
      case "270":
        height = this.props.width;
        width = this.props.height;
        className = "gb270";
        marginTop = height;
        pencilTop = `-${height}px`
        break;

      default:
        // "0"
        break;
    }
    var that = this;
    var hideBox = this.props.row === 0 && this.props.column === 3;
    var editMode = this.props.editMode ? (hideBox ? false : true) : false;
   
    return (
      <Subscribe>
        {context => (
          <div
            style={{
              position: "relative",
              border: "0px dashed red",
              width,
              height
            }}
            onClick={() => {
              if (onGroupClicked) onGroupClicked(context, this);
            }}
          >
            <div
              className={className}
              style={{
                display: "flex",
                justifyContent: "center",
                alignItems: "center",
                marginLeft: marginLeft,
                marginTop,
                border: editMode ? "1px dashed #aaaaaa" : null,
                padding: editMode ? "0px" : "1px",
                fontSize: this.props.fontsize,
                lineHeight: this.props.lineheight,
                backgroundColor: this.props.color,
                color: this.props.textColor ? this.props.textColor : "#000000",
                height: this.props.height,
                width: this.props.width,
                cursor: editMode ? "pointer " : "default"
              }}
            >
              <div className="gbText">
                {this.props.children ? this.props.children : this.props.text}
              </div>
              {editMode &&
                this.state.showProperties &&
                this.state.editorClicked &&
                context.editor.groupProperties({ ...this.props }, () => {
                  this.setState({ showProperties: false });
                })}
            </div>
            {context && editMode && this.state.editorClicked && context.editor && (
              <div
                onClick={(e, c) => {
                  e.stopPropagation();

                  that.setState({ showProperties: !that.state.showProperties });
                }}
                style={{
                  top: pencilTop,
                  left: "3px",
                  position: "absolute",
                  cursor: "pointer"
                }}
                className="groupboxeditor"
              >
                <i
                  class={
                    this.state.showProperties
                      ? "ms-Icon ms-Icon--EditSolid12"
                      : "ms-Icon ms-Icon--Edit"
                  }
                  aria-hidden="true"
                />
              </div>
            )}
          </div>
        )}
      </Subscribe>
    );
  }
}
