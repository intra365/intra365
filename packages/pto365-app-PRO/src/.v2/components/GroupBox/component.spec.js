import React, { Component } from 'react';

import GroupBox from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<GroupBox about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)