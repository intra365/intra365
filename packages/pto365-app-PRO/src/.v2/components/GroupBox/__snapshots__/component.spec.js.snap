// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupBox
    about="🍕"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="gb0"
        style={
                Object {
                        "alignItems": "center",
                        "backgroundColor": undefined,
                        "color": "#000000",
                        "cursor": "default",
                        "display": "flex",
                        "fontSize": undefined,
                        "height": undefined,
                        "justifyContent": "center",
                        "lineHeight": undefined,
                        "marginLeft": 0,
                        "marginTop": 0,
                        "width": undefined,
                      }
        }
>
        <div
                className="gbText"
        />
</div>,
      "style": Object {
        "border": "0px dashed red",
        "height": undefined,
        "width": undefined,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="gbText"
/>,
        "className": "gb0",
        "style": Object {
          "alignItems": "center",
          "backgroundColor": undefined,
          "color": "#000000",
          "cursor": "default",
          "display": "flex",
          "fontSize": undefined,
          "height": undefined,
          "justifyContent": "center",
          "lineHeight": undefined,
          "marginLeft": 0,
          "marginTop": 0,
          "width": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "gbText",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="gb0"
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": undefined,
                              "color": "#000000",
                              "cursor": "default",
                              "display": "flex",
                              "fontSize": undefined,
                              "height": undefined,
                              "justifyContent": "center",
                              "lineHeight": undefined,
                              "marginLeft": 0,
                              "marginTop": 0,
                              "width": undefined,
                            }
          }
>
          <div
                    className="gbText"
          />
</div>,
        "style": Object {
          "border": "0px dashed red",
          "height": undefined,
          "width": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="gbText"
/>,
          "className": "gb0",
          "style": Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "color": "#000000",
            "cursor": "default",
            "display": "flex",
            "fontSize": undefined,
            "height": undefined,
            "justifyContent": "center",
            "lineHeight": undefined,
            "marginLeft": 0,
            "marginTop": 0,
            "width": undefined,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "gbText",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
